File: Auction/src/main/java/longcat/auction/src/AuctionCatalogue.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package longcat.auction.src;

import java.io.Serializable;
import java.util.HashMap;
import java.util.List;
import longcat.auction.utils.AbstractDAO;

/**
 * Container class for Auction objects
 * @author emesven
 */
public final class AuctionCatalogue extends AbstractDAO<AuctionObject, Long> implements Serializable {

    public static AuctionCatalogue getDefault() {
        return new AuctionCatalogue(("auction_pu"));
    }

    public AuctionCatalogue(String puName) {
        super(AuctionObject.class, puName);
    }

    public List<AuctionObject> getAllAuctions() {
        return getRange(0, getCount());
    }
}

File: AuctionAdmin/src/main/webapp/js/core/auctionCatalogue.js
/* 
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

var AuctionCatalogue = function(baseUri) {
    this.baseUri = baseUri;
};

AuctionCatalogue.prototype = (function() {

    return {
        
        getRange: function(first, last) {
            return $.getJSON(this.baseUri + "/range?" + "first=" + first + "&" + "last=" + last);
        },
        
        getCount: function() {
            return $.getJSON(this.baseUri + "/count");
        },

        update: function(auctionobject) {
            return $.ajax({
                type: 'PUT',
                url: this.baseUri + "/" + auctionobject.id ,
                data: "name=" + auctionobject.name + "&info=" + auctionobject.info +
                        "&price=" + auctionobject.price
            });
        },
                
        remove: function(id) {
            return $.ajax({
                type: 'DELETE',
                url: this.baseUri + "/" + id
            });
        }
    };
}());
